src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1051,7 +1052,8 @@ private async ValueTask RefreshInstalledAndUpdatesTabsAsync()
             IInstalledAndTransitivePackages installedAndTransitivePackages = await PackageCollection.GetInstalledAndTransitivePackagesAsync(loadContext.ServiceBroker, loadContext.Projects, includeTransitiveOrigins: true, token);
             installedPackageCollection = PackageCollection.FromPackageReferences(installedAndTransitivePackages.InstalledPackages);
             PackageCollection transitivePackageCollection = PackageCollection.FromPackageReferences(installedAndTransitivePackages.TransitivePackages.Where(p => p.TransitiveOrigins.Any()));
-            IEnumerable<PackageVulnerabilityMetadataContextInfo>[] transitivePackageVulnerabilities = await Task.WhenAll(transitivePackageCollection.Select(p => _packageVulnerabilityService.GetVulnerabilityInfoAsync(p, token)));
+            _packageVulnerabilityService.Refresh();
+            IEnumerable<PackageVulnerabilityMetadataContextInfo>[] transitivePackageVulnerabilities = await Task.WhenAll(transitivePackageCollection.Select(_packageVulnerabilityService.GetVulnerabilityInfoAsync));
```

Comment:
We can do things more efficiently here. I think this was an issue in the previous implementation as well.

Say we have 10 packages. 
We create 10 tasks, 10 async state machines, that each call a method that awaits on an async lazy (thus potentially blocking all the tasks on the same thing). 
Once the "costly" fetching async work is done, all the necessary work is back to being synchronous. 

Parallelization is not super helpful when all tasks will be blocked on the same work happening. 
While async is great when there's potentially expensive work happening, it's not "free". 

The cheaper way to do this is, pass the service the list of packages and get a list of vulnerability results. 
That way you don't have the Task.WhenAll allocations and the extra state machine allocations.

Reply:
To add to that, should `PackageVulnerabilityService` be redesigned a bit? I'm not following the concept of "GetAll" versus "Get" which both seem to await on the same thing. As a consumer, is it clear that "Get" is triggering the "GetAll" implicitly?